tests/test_sanity.py

```diff
@@ -584,13 +576,12 @@ def test_emconfig(self):
 
     temp_dir = tempfile.mkdtemp(prefix='emscripten_temp_')
 
-    os.chdir(temp_dir)
-    self.do([PYTHON, EMCC, '--em-config', custom_config_filename] + MINIMAL_HELLO_WORLD + ['-O2'])
-    result = run_js('a.out.js')
-    
+    with chdir(temp_dir):
+      self.do([PYTHON, EMCC, '--em-config', custom_config_filename] + MINIMAL_HELLO_WORLD + ['-O2'])
+      result = run_js('a.out.js')
+
     # Clean up created temp files.
     os.remove(custom_config_filename)
-    os.chdir(path_from_root())
```

Comment:
This one just wasn't even trying to clean up was it
So if the emcc invocation failed, would the chdir(temp_dir) be persisted across tests, or is there some other mechanism to keep things from getting that out of hand?
Either way this is better